### ISSUE TITLE
Application Search broken with unexpected non-whitespace character

### ISSUE THREAD

**Issue by hanchenlu** [oliverschwendener/ueli#967]
I recently upgraded to release 8.23.1 from 8.22.1 and Application Search was broken with the following error. I attempted a re-install, but the issue has persisted, though I may not have been thorough enough in wiping previous config files and application data.

![image](https://user-images.githubusercontent.com/49032750/202004801-9d309dff-27be-42c5-a28a-83c35e0087da.png)

Relevant section from the debug log

```
2022-11-14T05:36:22.103Z error: Unexpected non-whitespace character after JSON at position 28825
SyntaxError: Unexpected non-whitespace character after JSON at position 28825
    at JSON.parse (<anonymous>)
    at C:\Program Files\ueli\resources\app.asar\bundle\main.js:8:803400
    at process.processTicksAndRejections (node:internal/process/task_queues:96:5)
2022-11-14T05:36:22.104Z error: Unexpected non-whitespace character after JSON at position 10513
SyntaxError: Unexpected non-whitespace character after JSON at position 10513
    at JSON.parse (<anonymous>)
    at C:\Program Files\ueli\resources\app.asar\bundle\main.js:8:890725
    at process.processTicksAndRejections (node:internal/process/task_queues:96:5)
```

**Comment by hanchenlu** [comment]
Issue seems to have resolved itself recently after some restarts from other system updates. Namely Steam, NVidia video and audio drivers, and K-Lite Codec Pack, none of which I figure would impact the functioning of Ueli.

**Comment by hanchenlu** [comment]
Issue seems to have cropped up again. Not sure what's going on. Seems to happen right after a restart due to updating NVidia graphics driver

```
SyntaxError: Unexpected non-whitespace character after JSON at position 10513
    at JSON.parse (<anonymous>)
    at C:\Program Files\ueli\resources\app.asar\bundle\main.js:8:890725
    at process.processTicksAndRejections (node:internal/process/task_queues:96:5)
2022-12-01T23:38:39.400Z debug: Check for updates
2022-12-01T23:38:40.033Z debug: Update check result: update not available
2022-12-01T23:41:34.184Z debug: Check for updates
2022-12-01T23:41:34.839Z debug: Update check result: update not available
2022-12-01T23:42:03.917Z debug: Successfully refresh index of plugin application-search-plugin
2022-12-01T23:42:05.882Z error: Unexpected non-whitespace character after JSON at position 48713
SyntaxError: Unexpected non-whitespace character after JSON at position 48713
    at JSON.parse (<anonymous>)
    at C:\Program Files\ueli\resources\app.asar\bundle\main.js:8:803400
    at process.processTicksAndRejections (node:internal/process/task_queues:96:5)
```

**Comment by tkohlmeier** [comment]
This might be caused by a special character in a path or file name found by the Application Search plugin. To check this, please run the following command in a command window (cmd.exe):

    powershell.exe -NonInteractive -NoProfile -Command "& { [Console]::OutputEncoding = [Text.UTF8Encoding]::UTF8; $ErrorActionPreference = 'SilentlyContinue'; $applications = New-Object Collections.Generic.List[String]; $extensions = New-Object System.Collections.Generic.HashSet[string] ([string[]]@('.lnk','.appref-ms','.url','.exe'), [System.StringComparer]::OrdinalIgnoreCase); 'C:\ProgramData\Microsoft\Windows\Start Menu\Programs','C:\Users\Torben\AppData\Roaming\Microsoft\Windows\Start Menu','C:\Users\Torben\Desktop','D:\Dropbox\Tools' | %{ Get-ChildItem -LiteralPath $_ -Recurse -File | Where-Object { $extensions.Contains($_.Extension) } | ForEach-Object { $applications.Add($_.FullName) } }; $result = (@{ errors = @($error | ForEach-Object { $_.Exception.Message }); applications = $applications } | ConvertTo-Json); Write-Host $result; }" > applications.json

This will create a file `applications.json` in the current directory. Now open this file with [Visual Studio Code](https://code.visualstudio.com/download) and press F8 (or click on Go -> Next Problem in the menu bar). The cursor should now jump to the position which causes the problem and show the reason. If you provide us with the details we should be able to fix this.

**Comment by hanchenlu** [comment]
Okay, so right after you replied the issue seemed to have resolved itself again, but just today it re-appeared. I ran your script with the file paths changed to the ones I use in my configuration, and the only line that looks out of place is at the end with `[0x7FF99712F0F0] ANOMALY: meaningless REX prefix used`. I was working on updating and tweaking my computer when it once again spontaneously resolved itself, and I ran that script again to compare the two application.json files. Both are identical except for that anomaly at the end of it when the application search was acting up. Don't know if you want me to post the file or the contents to help diagnose, but that final line is the only indication that anything is amiss.
